addons/mail/models/mail_thread.py

```diff
@@ -1181,7 +1183,10 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
 
                 mail_message = self.env['mail.message']
                 if email_part:
-                    email = email_part.get_payload()[0]
+                    if email_part.is_multipart():
+                        email = email_part.get_payload()[0]
+                    else:
```

Comment:
similarly, the other case could be made explicit, because you're not even testing the type of payload:
```suggestion
                    elif email_part.get_content_type() == 'text/rfc822-headers':
                        # text/rfc822-headers is a str that needs to be decoded
```